packages/dx-react-grid/docs/guides/grouping.md

```diff
@@ -43,3 +43,26 @@ In the following example the Grid functions in the [uncontrolled mode](controlle
 In the [controlled mode](controlled-and-uncontrolled-modes.md), pass a grouping options array to the `GroupingState` plugin's `grouping` property and handle the `onGroupingChange` event to control the grouping state.
 
 .embedded-demo(grouping/local-grouping-controlled)
+
+## Custom Grouping Values
+
+If you want to provide a custom group value for a column, pass your grouping function to the `LocalGrouping` plugin’s `getGroupValue` property. Usually, when using a custom grouping function, you want to see the grouped column in the table (by default it is hidden when grouped). To do this, add the `showWhenGrouped` field to the particular column configuration object and set its value to `true`. For instance, in the following example, we use the first letter of the `city` column as a group value and keep this column visible in the table.
+
+.embedded-demo(grouping/local-grouping-custom)
+
+For more complex scenarios, you can also specify a custom function in the `showColumnWhenGrouped` property of the `TableGroupRow` plugin to save the grouped columns in the table:
+
+```js
+  <Grid>
+    <GroupingState
+      grouping={[{ columnName: 'city' }, { columnName: 'car' }]}
+    />
+    <LocalGrouping />
+    <TableView />
+    <TableHeaderRow showColumnWhenGrouped={columnName => columnName === 'city' || columnName === 'car'}/>
+    <TableGroupRow />
+  </Grid>
+```
+For complex groupings scenarios, pass your grouping function to the `LocalGrouping` plugin’s `getGroupKey` property. This function should return the custom key value by which the rows will be grouped. In the following example, all records are divided into two groups by the first letter of the `name` column. Also the `groupCellTemplate` property of the `TableGroupRow` is used to provide a custom appearance for the group rows.
```

Comment:
Write the first sentences like this:
When a returned value from the `getGroupValue` is not primitive, it is needed to define a custom `getGroupKey` function. This function is needed to get a string from a non-primitive value returned by the `getGroupValue` function in order to check groups equality. In other words, the `getGroupKey` function is a hash function used to identify equality of complex objects.